funcr/funcr.go

```diff
@@ -341,6 +341,17 @@ const (
 	flagRawStruct = 0x1 // do not print braces on structs
 )
 
+func isNil(i interface{}) bool {
```

Comment:
Is this faster than wrapping the MarshalLog and String invocations in a function which recovers from a failure inside the called function?

That's how Dims solved it in klog:
https://github.com/kubernetes/klog/commit/125dec9bbe623ab72ec25154974434e2ffc28782#diff-d11e021bcd662e6127823cd93eeac7a9aec303039137c2f27974cff399142e84

It has two advantages:
- if the interface contains a valid pointer to a struct but the called function then crashes for some other reason (nil access for a field in that struct, for example), then the recovery mechanism catches that, too
- if the function implementation itself handles nil gracefully, then logging keeps using that fallback instead of making up its own